tests/testsuite/freshness.rs

```diff
@@ -492,62 +510,86 @@ fn changing_bin_features_caches_targets() {
         .build();
 
     p.cargo("build")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([..])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
+        .run();
+    p.rename_run("foo", "off1")
+        .with_stdout_data(str![[r#"
+feature off
+
+"#]])
         .run();
-    p.rename_run("foo", "off1").with_stdout("feature off").run();
 
     p.cargo("build --features foo")
-        .with_stderr(
-            "\
-[COMPILING] foo v0.0.1 ([..])
-[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
-",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
+        .run();
+    p.rename_run("foo", "on1")
+        .with_stdout_data(str![[r#"
+feature on
+
+"#]])
         .run();
-    p.rename_run("foo", "on1").with_stdout("feature on").run();
 
     /* Targets should be cached from the first build */
 
     let mut e = p.cargo("build -v");
 
     // MSVC does not include hash in binary filename, so it gets recompiled.
     if cfg!(target_env = "msvc") {
-        e.with_stderr(
-            "\
-[DIRTY] foo v0.0.1 ([..]): the list of features changed
-[COMPILING] foo[..]
-[RUNNING] `rustc [..]
-[FINISHED] `dev`[..]",
-        );
+        e.with_stderr_data(str![[r#"
+[DIRTY] foo v0.0.1 ([ROOT]/foo): the list of features changed
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name [..]
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
```

Comment:
ok this one is even more surprising 🤯 

https://github.com/rust-lang/cargo/actions/runs/9729712870/job/26851897369#step:11:4057

```diff
 thread 'freshness::changing_bin_features_caches_targets' panicked at tests\testsuite\freshness.rs:561:7:

---- expected: tests\testsuite\freshness.rs:546:28
++++ actual:   stderr
-  1      - [DIRTY] foo v0.0.1 ([ROOT]/foo): the list of features changed
-  2      - [COMPILING] foo v0.0.1 ([ROOT]/foo)
-  3      - [RUNNING] `rustc --crate-name [..]
+       1 + [FRESH] foo v0.0.1 ([ROOT]/foo)
   4    2 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
```
How could MSVC suddenly get this behavior change, I don't think there is any logical change on `changing_bin_features_caches_targets`. 

It was still `[DIRTY]` on 6/28: https://github.com/rust-lang/cargo/actions/runs/9700016140/job/26770429920#step:11:4070

But since yesterday it became `[FRESH]`: https://github.com/rust-lang/cargo/actions/runs/9724808711/job/26841405762#step:11:4057

I did [rebase](https://github.com/rust-lang/cargo/pull/14161#event-13339691283) onto `main` when squashing yesterday, but there was no new change other than tests from `main` according to [diff](https://github.com/rust-lang/cargo/compare/cbf4b1bf5cd5fc45719be5ab4231ae8b77813ec9..56ca35f7a65bbed6f75b5cf973f729a2ab0be328).

Reply:
allow me to push a few temporary commits to test this out ⌛

Reply:
Did you know on which toolchain version that the behavior changes?

Reply:
> It was still `[DIRTY]` on 6/28: [Run 9700016140](https://github.com/rust-lang/cargo/actions/runs/9700016140/job/26770429920#step:11:4070)

Sorry I was wrong about this. `changing_bin_features_caches_targets` has been failing has been fail since beginning. It just failed in earlier assertion 🤦.

Reverting my change to `changing_bin_features_caches_targets` actually fix the test (c826531 and its [ci](https://github.com/rust-lang/cargo/actions/runs/9747539766/job/26900484318#step:11:2240)), so there must be something incorrect from my change. Let me check this later.

Reply:
So this was just from a silly typo. I accidentally added a duplicated `.run()` at L552. Removed in latest amended commit.